mqtt/policy/src/core/mod.rs

```diff
@@ -702,25 +709,130 @@ pub(crate) mod tests {
         assert_matches!(policy.evaluate(&request), Ok(Decision::Allowed));
     }
 
-    /// `TestSubstituter` replaces any value with the corresponding identity or resource
+    /// Scenario:
+    /// - Have a policy with a custom identity matcher
+    /// - Have two variable rules (deny and allow) for an identity, such that
+    ///   both rules match a given request identity.
+    /// - But the two rules must be different in resources.
```

Comment:
I think we need to say after this line something like:
"We make a request to the allowed resource"

It isn't clear reading this what the resource requested is.